datasource/src/main/scala/quasar/physical/blobstore/BlobstoreDatasource.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.physical.blobstore
+
+import slamdata.Predef._
+import quasar.api.QueryEvaluator
+import quasar.api.datasource.DatasourceType
+import quasar.api.resource.{ResourceName, ResourcePath, ResourcePathType}
+import quasar.blobstore.Blobstore
+import quasar.connector.{MonadResourceErr, ResourceError}
+import quasar.connector.datasource.LightweightDatasource
+import quasar.contrib.scalaz.MonadError_
+import quasar.contrib.std.errorNotImplemented
+
+import cats.Applicative
+import cats.effect.IO
+import cats.syntax.applicative._
+import cats.syntax.option._
+import fs2.Stream
+import qdata.QDataEncode
+
+class BlobstoreDatasource[F[_]: Applicative: MonadResourceErr](
+  val kind: DatasourceType,
+  blobstore: Blobstore[F])
+  extends LightweightDatasource[F, Stream[F, ?]] {
+
+  override def evaluator[R: QDataEncode]: QueryEvaluator[F, ResourcePath, Stream[F, R]] =
+    errorNotImplemented
+
+  override def pathIsResource(path: ResourcePath): F[Boolean] =
+    errorNotImplemented
+
+  override def prefixedChildPaths(prefixPath: ResourcePath)
+      : F[Option[Stream[F, (ResourceName, ResourcePathType)]]] =
+    blobstore.list(prefixPath).some.pure[F]
```

Comment:
The `Blobstore#list` signature is forcing implementations to push "not found" errors into the stream defeating the purpose of the `prefixedChildPaths` return type, which is to allow for surfacing this error before streaming is started.

Reply:
Yeah, I did that to stay close to the interface of `fs2-blobstore` and because I think for Azure this is what we want anyway (see https://github.com/slamdata/quasar-blobstore/pull/1#discussion_r229211372 ). 

But I'm currently unsure if we still can/should integrate with `fs2-blobstore`, depends a bit to what extent our goals can be unified with theirs.